### PR TITLE
Env refactor: rename $cur__source_root

### PR DESCRIPTION
**Pull request by andreypopp**
TODO:
* [x] Merge #23 first

Previously we had `$cur__source_root` which pointed to the original source directory of a package. We rename it to `$esy_build__source_root` as it is an implementation detail and not part of PJC spec.

### CODE REVIEW

File: tests/TestOCamlCompilation/__snapshots__/test.js.snap [hunk 1]
```diff
@@ -15,9 +15,9 @@ exports[`test build 1`] = `
 `;
 
 exports[`test build 2`] = `
-"Hello, I\'m PackageC!!!!
```
Comment (andreypopp) [inline tests/TestOCamlCompilation/__snapshots__/test.js.snap:18]: Hm...

File: tests/TestDepBuildsInSource/__snapshots__/test.js.snap [hunk 2]
```diff
@@ -56,7 +54,6 @@ export dep__etc=\"$dep__install/etc\"
 # [BuiltIns]
 export testdepoutofsourcebuildisfalse__name=\"TestDepOutOfSourceBuildIsFalse\"
 export testdepoutofsourcebuildisfalse__version=\"1.0.0\"
-export testdepoutofsourcebuildisfalse__source_root=\"$esy__sandbox\"
```
Comment (jordwalke) [inline tests/TestDepBuildsInSource/__snapshots__/test.js.snap:59]: what logic change causes this to no longer be output?

Comment (andreypopp) [reply tests/TestDepBuildsInSource/__snapshots__/test.js.snap:59]: See https://github.com/jordwalke/esy/pull/24/files#diff-57f4be5f4025a4fd81a3bbc2cdfc2194L192

We discussed in Discord that `$*__source_root` is an is an impl. detail so I moved it to `esyBuildEjectCommand.js` under `$esy_build__source_root` name.

Comment (jordwalke) [reply tests/TestDepBuildsInSource/__snapshots__/test.js.snap:59]: Oh, I see I just didn't originally understand that now `esy_build__source_root` will only apply to the rootmost package. I suppose it makes sense. I don't see many use cases for a "cur" version.

**Pull request merged**